src/utils/url/getAbsoluteUrl.ts

```diff
@@ -8,8 +8,8 @@ export function getAbsoluteUrl(path: string, baseUrl?: string): string {
   }
 
   // Resolve a relative request URL against a given custom "baseUrl"
-  // or the current location (in the case of browser/browser-like environments).
-  const origin = baseUrl || (typeof location !== 'undefined' && location.origin)
+  // or the document baseURI (in the case of browser/browser-like environments).
+  const origin = baseUrl || document.baseURI
```

Comment:
Let's keep the `typeof` guard but check the existence of the `document` instead:

```suggestion
  const origin = baseUrl || (typeof document !== 'undefined' && document.baseURI)
```

This code also runs in Node.js, where `document` won't exist. The guard is a great way to fall back to using `document.baseURI` whenever there _is_ a document (which may be the case in Node.js if you're using browser polyfills like jsdom).

Reply:
I will update the PR now. Thanks!